src/AWS4AuthRequest.jl

```diff
@@ -90,10 +101,14 @@ function sign_aws4!(method::String,
     end
 
     # Sort and lowercase() Headers to produce canonical form...
-    canonical_headers = sort!(map(headers) do (k, v)
-        string(lowercase(k), ':', replace(strip(v), r"\s+" => " "))
-    end)
-    signed_headers = join(sort!(map(lowercase∘first, headers)), ";")
+    unique_header_keys = sort!(unique!(map(lowercase∘first, headers)))
+    canonical_headers = map(unique_header_keys) do k
+        # All values present for this key (case-insensitive) with whitespace normalized
+        vs = map(i->_normalize_ws(last(headers[i])),
+                 findall(x->lowercase(first(x)) == k, headers))
+        string(k, ':', join(vs, ','))
+    end
```

Comment:
Alternative:
```julia
using DataStructures: DefaultOrderedDict
normalized_headers = DefaultOrderedDict{String,Vector{String}}(() -> String[])
for k in sort!(collect(keys(headers)))
    ck = lowercase(k)
    push!(canonical_headers[ck], _normalize_ws(header[k]))
end
canonical_headers = [string(k, ':', join(vs, ',')) for (k, vs) in normalized_headers]
unique_header_keys = collect(keys(normalized_headers))
```
or for dependency haters:
```julia
unique_header_keys = Vector{String}()
normalized_headers = Dict{String,Vector{String}}()
for k in sort!(collect(keys(headers)))
    ck = lowercase(k)

    if !haskey(canonical_headers, ck)
        canonical_headers[ck] = Vector{String}()
    end

    push!(canonical_headers[ck], _normalize_ws(header[k]))
    push!(normalized_headers, ck)
end
canonical_headers = [string(k, ':', join(vs, ',')) for (k, vs) in normalized_headers]
```
The main advantage to this approach is you avoid some iterations

Reply:
I had to revert this change, as it did not produce the correct result.

Reply:
All is well now